Assets/FruitNodeController.cs

```diff
@@ -0,0 +1,76 @@
+using Fusion;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class FruitNodeController : NetworkBehaviour
+{
+    private struct SpawnerLocation
+    {
+        public SpawnerLocation(Vector3 position, Quaternion rotation, bool taken, NetworkObject obj)
+        {
+            this.position = position;
+            this.rotation = rotation;
+            this.taken = taken;
+            this.spawner = obj;
+        }
+
+        public Vector3 position;
+        public Quaternion rotation;
+        public bool taken;
+        public NetworkObject spawner;
+    }
+
+    [Tooltip("Gameobjects representing places where fruit can grow")]
+    public GameObject[] nodes;
+
+    private SpawnerLocation[] nodesAvailability;
+
+    [Tooltip("Spawner Gameobject")]
+    public NetworkPrefabRef Spawner;
+
+    public override void Spawned()
+    {
+        if (nodes.Length < 1)
+        {
+            Debug.LogWarning("Stages array in this plant is empty. You forgot me.");
+            return;
+        }
+
+        nodesAvailability = new SpawnerLocation[nodes.Length];    
+
+        for (int i = 0; i < nodes.Length; i++)
+        {
+            nodes[i].SetActive(false);
+
+            nodesAvailability[i] = new SpawnerLocation(nodes[i].transform.position, nodes[i].transform.rotation, false, null);
+        }
+
+        if (Object.HasStateAuthority)
+            LightingManager.Get().sunUp.AddListener(activateRandomNode);
+    }
+
+    private void activateRandomNode()
+    {
+        if (!gameObject.activeInHierarchy)
+            return;
+
+        for (int i = 0; i < nodesAvailability.Length; i++)
+        {
+            if (nodesAvailability[i].spawner == null)
+            {
+                nodesAvailability[i].taken = false;
+            }
+        }
+
+        var randomIndex = Random.Range(0, nodes.Length - 1);
+
+        if (!nodesAvailability[randomIndex].taken)
+        {
+            nodesAvailability[randomIndex].taken = true;
+            var networkObj = Runner.Spawn(Spawner, nodesAvailability[randomIndex].position, nodesAvailability[randomIndex].rotation);
+            nodesAvailability[randomIndex].spawner = networkObj;
+            networkObj.transform.parent = transform;
+        }

```

Comment:
interesting that the effects of this impl are:

- a fruit tries to grow every day
- if the fruit tries to grow on a taken slot, no fruit today

this ends up making it not really "another fruit every day", but "often another fruit", which is cool. But I also think we'll want to make it so that fruit grows "every few days", and also possibly that the odds of a fruit growing are independent of the # of fruit slots taken up. That's all just food for thought though!

Reply:
I think I'm thinking of eventually something like a "plont listener gives it someNum plont points, then plont decides whether and how to spend those points, e.g. on a) new limbs b) deeper roots c) new flowers d) new fruit e) fruit maturation" model, where someNum could also be random + a computation based on the plont's attributes and things like season, sunlight, elevation, soil condition, etc